python/ray/serve/request_params.py

```diff
@@ -42,7 +42,7 @@ def adjust_relative_slo_ms(self) -> float:
         return current_time_ms + slo_ms
 
     def ray_serialize(self):
-        return pickle.dumps(self.__dict__, protocol=5)
+        return pickle.dumps(self.__dict__)
```

Comment:
This looks fine for now, but how about we just remove the `async_future` object from the `Query` entirely so we don't need to define the custom serializer?